src/pages/resultsView/coExpression/CoExpressionPlot.tsx

```diff
@@ -0,0 +1,636 @@
+
+    private svgRef(svg:SVGElement|null) {
+        this.svg = svg;
+    }
+
+    private onButtonClick(event:React.MouseEvent<HTMLButtonElement>) {
+        switch ((event.target as HTMLButtonElement).name) {
+            case EventKey.downloadSVG:
+                if (this.svg) {
+                    this.svgsaver.asSvg(this.svg, "coexpression.svg");
+                }
+                break;
+            case EventKey.downloadPDF:
+                if (this.svg) {
+                    svgToPdfDownload("coexpression.pdf", this.svg);
+                }
+                break;
+        }
+    }
+
+    private makeMouseEvents() {
+        let disappearTimeout:Timer | null = null;
+        const disappearDelayMs = 250;
+
+        return [{
+            target: "data",
+            eventHandlers: {
+                onMouseOver: () => {
+                    return [
+                        {
+                            target: "data",
+                            mutation: (props: any) => {
+                                this.tooltipModel = props;
+                                this.pointHovered = true;
+
+                                if (disappearTimeout !== null) {
+                                    clearTimeout(disappearTimeout);
+                                    disappearTimeout = null;
+                                }
+
+                                return { active: true };
+                            }
+                        }
+                    ];
+                },
+                onMouseOut: () => {
+                    return [
+                        {
+                            target: "data",
+                            mutation: () => {
+                                if (disappearTimeout !== null) {
+                                    clearTimeout(disappearTimeout);
+                                }
+
+                                disappearTimeout = setTimeout(()=>{
+                                    this.pointHovered = false;
+                                }, disappearDelayMs);
+
+                                return { active: false };
+                            }
+                        }
+                    ];
+                }
+            }
+        }];
+    }
+
+    @computed get xScale() {
+        return this.props.xLogScale ? "log" : "linear";
+    }
+
+    @computed get yScale() {
+        return this.props.yLogScale ? "log" : "linear";
+    }
+
+    @computed get plotData() {
+        return computePlotData(
+            this.props.data,
+            0.05,
+            !!this.props.xLogScale,
+            !!this.props.yLogScale
+        );
+    }
+
+    @computed get plotBounds() {
+        return {
+            x: [0, this.plotData.max.x + this.plotData.xPadding],
+            y: [0, this.plotData.max.y + this.plotData.yPadding]
+        };
+    }
+
+    @computed get stroke() {
+        if (this.props.showMutations) {
+            return (d:{mutX:boolean, mutY:boolean})=>{
```

Comment:
stroke and fill seems pretty prime for unit test.   would be expensive to test this with e2e, no?

Reply:
wasnt planning on testing it e2e? I think its too simple for unit test, but whole thing would just be captured by screenshots